[docs] Add basic build-script command for Linux and corelibs flag (#38067)
swiftlang/swift: docs/HowToGuides/GettingStarted.md

       --sccache --release-debuginfo --swift-disable-dead-stripping --test \
       --xcode
     ```
   Linux (uses Ninja):
     ```sh

>      utils/build-script --release-debuginfo --test --skip-early-swift-driver

Reviewer: What is the reason for recommending `--skip-early-swift-driver` here?

Reviewer 2: Building the Swift Driver for the compiler instead of the C++ Driver was just enabled last month in #37122. I think it requires having a whole Swift toolchain for the platform already installed, though I haven't tried it.

I'm just skipping it for now because I think it requires additional instructions on installing a host Swift toolchain and I haven't tried it, but we should enable it at some point.

Reviewer 2: Alright, I just tried it again without that flag. With a Swift toolchain installed in `/usr/bin` in my Fedora Core 33 VPS, it built fine. If I removed that toolchain package, it spits out this warning and continues to build cmark:
```
-- Warning: {} Host toolchain could not locate a compiler to build swift-driver. (Try `--skip-early-swift-driver`)
```
I think it used to fail before, which is why I started using that flag.

I think it's better not to confuse noobs with that warning, but it appears to still work when the swift driver is not built early, so we could remove that flag now.

Reviewer: Makes sense. Given the warning, I agree it's better to have the `--skip` flag.

     ```
   This will create a directory
   `swift-project/build/Ninja-RelWithDebInfoAssert`
   (with `Xcode` instead of `Ninja` if you used `--xcode`)

>    containing the Swift compiler and standard library build artifacts.

Reviewer: There are more build artifacts, including Clang and the LLVM tools. I think it would be better to not specify more details here.

Reviewer 2: I think it's better to specify exactly what this first basic command builds for noobs, so I added clang/LLVM to the list instead. Let me know what you think.

Reviewer: That's also reasonable.

>    containing the Swift compiler, clang/LLVM, and standard library build artifacts.

Reviewer: ```suggestion
   containing the build artifacts for the Swift compiler, the Swift standard
   library, Clang and LLVM.
```
Having LLVM in-between two Swift-related projects seems a bit odd order-wise.

Reviewer 2: Changed the order.

   - If the build fails:
     See [Troubleshooting build issues](#troubleshooting-build-issues).

>    If you would like to build the Swift corelibs, ie swift-corelibs-libdispatch,
>    swift-corelibs-foundation, and swift-corelibs-xctest, also on Linux, add the
>    `--xctest` flag to `build-script`.

Reviewer: ```suggestion
   If you would like to additionally build swift-corelibs-libdispatch,
   swift-corelibs-foundation, and swift-corelibs-xctest (these are particularly
   useful on Linux) add the `--xctest` flag to `build-script`.
```

Reviewer 2: Reworded a bit but [swift-corelibs-foundation](https://github.com/apple/swift/blob/main/utils/build-script-impl#L2317) and [swift-corelibs-libdispatch](https://github.com/apple/swift/blob/main/utils/build-script-impl#L2382) are not built on macOS, so kept the emphasis on this flag only being for Linux.

   - If the build fails:
     See [Troubleshooting build issues](#troubleshooting-build-issues).

   If you would like to build the Swift corelibs, ie libdispatch, Foundation,
   and XCTest, also on Linux, add the `--xctest` flag to `-build-script`.

> 

Reviewer: Could you instead spell out the full names `swift-corelibs-libdispatch` and so on? This wording is potentially misleading as it makes it sound like one is building Foundation etc. However, what one is actually building is swift-corelibs-foundation etc. This distinction matters because bugs for Foundation (in the Darwin SDKs) should preferably be reported through feedback assistant, whereas bugs for swift-corelibs-foundation should be reported through JIRA.

Reviewer 2: I was unaware of the naming distinction, done.
